homework-solutions/ts/hw-15/task-2.ts

```diff
@@ -0,0 +1,75 @@
+// 1. Define the IPerson interface
+// Contains personal info, experience, and a method to return details
+interface IPerson {
+    name: string;
+    surname: string;
+    experienceYears: number;
+
+    getDetails(): string;
+}
+
+// 2. Abstract class Employee implementing IPerson
+abstract class Employee implements IPerson {
+    name: string;
+    surname: string;
+    experienceYears: number;
+
+    protected salary: number;
+
+    constructor(name: string, surname: string, experienceYears: number) {
+        this.name = name;
+        this.surname = surname;
+        this.experienceYears = experienceYears;
+        this.salary = 0;
```

Comment:
И в классах наследниках можно не обьявлять данньій метод в консрукторе
```suggestion
        this.calculateSalary();
```